src/REVLoans.sol

```diff
@@ -308,43 +314,31 @@ contract REVLoans is ERC721, ERC2771Context, IREVLoans {
         external
         payable
         override
-        returns (uint256 newLoanId, REVLoan memory, REVLoan memory newLoan)
+        returns (uint256 refinancedLoanId, uint256 newLoanId, REVLoan memory, REVLoan memory newLoan)
```

Comment:
If we need `nonReentrant` on the other public functions we should add it here as well to be safe. Perhaps we can make `borrowFrom(..)` an external function that has the modifier and copy all its current logic to `_borrowFrom(..)` which we can also use here.

Reply:
true. im not yet convinced reentrency is needed cc @simplemachine92 do we have a proof?

Reply:
it was precautionary, but I'll confirm/deny in tests.